doc/lua_api.txt

```diff
@@ -1118,54 +1118,109 @@ A box of a regular node would look like:
 
     {-0.5, -0.5, -0.5, 0.5, 0.5, 0.5},
 
+Perlin noise
+------------
+Perlin noise creates a continuously-varying value depending on the input values.
+Usually in Minetest the input values are either 2D or 3D co-ordinates in nodes.
+The result is mostly used during mapgen to create the terrain shape, or to vary
+the density of decorations or vary the structure of ores.
+
+### Structure of perlin noise
+An 'octave' is a simple noise generator that outputs a value between -1 and 1.
+The smooth wavy noise it generates has a single characteristic scale, almost
+like a 'wavelength', so on its own does not create fine detail.
+So perlin noise combines several octaves to create variation on multiple scales.
+Each additional octave has a smaller 'wavelength' than the previous, often half
+the size of the previous (if `lacunarity` = 2.0, see below).
+
+This results in noise varying very roughly between -2.0 and 2.0, with an average
+value of 0, so `scale` and `offset` are then used to multiply and offset the
+noise variation. 
+
+The final perlin noise variation is created as follows:
+
+noise = offset + scale * (octave1 +
+                          octave2 * persistence +
+                          octave3 * persistence ^ 2 +
+                          octave4 * persistence ^ 3 +
+                          ...)
+
+(The word 'octave' is used because each additional simple noise generator often
+has a frequency double that of the previous, and in music an 'octave' is a
+doubling of frequency. Noise is analogous to musical tones and harmony and can
+be thought of as spatial music in higher dimensions.)
+
 Noise Parameters
 ----------------
 Noise Parameters, or commonly called "`NoiseParams`", define the properties of
 perlin noise.
 
 ### `offset`
-Offset that the noise is translated by (i.e. added) after calculation.
+After the multiplication by `scale`, this is added the the result and is the
```

Comment:
The order of explanation matches the order of parameters when defined. The order actually makes sense in terms of how noise is calculated because it goes upwards from (spread, seed, octaves, persist, lacunarity, flags) to scaling by (scale) then finally offsetting by (offset).

Reply:
Your call, but it really doesn't help when trying to understand it.